src/fromager/gitutils.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import pathlib
+from urllib.parse import urlparse
+
+from packaging.requirements import Requirement
+
+from fromager import context, external_commands
+
+logger = logging.getLogger(__name__)
+
+
+def git_clone(
+    *,
```

Comment:
Clarification question / comment: Can you help me understand why do we need to pass `*` as first argument?

Reply:
In python this syntax means that a caller of the function must name all of the arguments (`func(arg1=val1, arg2=val2)`) , rather than passing them by position (`func(val1, val2)`). I copied this from a utility function in our downstream repo and just kept this aspect, but it's not integral to how the function works.

Reply:
Okay, makes sense. Thank you!